### ISSUE TITLE
Deprecate twisted.python.reflect.funcinfo

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2079]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#2079|
|Type|task|
|Created|2006-09-11 22:52:45Z|

It's broken and worthless. Even the docstring says so.

<details><summary>Searchable metadata</summary>

```
trac-id__2079 2079
type__task task
reporter__radix radix
priority__highest highest
milestone__Core_2_5 Core-2.5
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1158015165000000 1158015165000000
changetime__1160582154000000 1160582154000000
version__None None
owner__ 
cc__radix cc__exarkun cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_1"></a>|@radix commented|
|-|-|

Review

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml set owner to @radix|
|-|-|


Looks good. Please merge.

Also, please delete Twisted/branches/radix :)

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_4"></a>|@radix commented|
|-|-|

where the hell did the patch go

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @exarkun|
|-|-|

trac is extra buggy, I guess.  Here's another try:

```
Index: twisted/python/reflect.py
===================================================================
--- twisted/python/reflect.py   (revision 18416)
+++ twisted/python/reflect.py   (working copy)
@@ -22,6 +22,7 @@
 import traceback
 import weakref
 import re
+import warnings
 
 RegexType = type(re.compile(""))
 
@@ -251,6 +252,10 @@
     """
     this is more documentation for myself than useful code.
     """
+    warnings.warn(
+        "[v2.5] Use inspect.getargspec instead of twisted.python.reflect.funcinfo",
+        DeprecationWarning,
+        stacklevel=2)
     code=function.func_code
     name=function.func_name
     argc=code.co_argcount
```

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_6"></a>|@ralphm set owner to @radix|
|-|-|

Looks fine (still?). Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [18437]) Apply funcinfo deprecation patch

Author: exarkun (radix)
Reviewer: ralphm (jml)
Fixes [#2079](https://github.com/twisted/twisted/issues/2079)

Deprecate the mostly broken and mostly useless funcinfo utility function.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
